xbmc/pvr/windows/GUIWindowPVRChannels.h

```diff
@@ -29,14 +29,14 @@ namespace PVR
     CGUIWindowPVRChannels(bool bRadio);
     virtual ~CGUIWindowPVRChannels(void) {};
 
-    bool OnMessage(CGUIMessage& message);
-    void GetContextButtons(int itemNumber, CContextButtons &buttons);
-    bool OnContextButton(int itemNumber, CONTEXT_BUTTON button);
-    bool Update(const std::string &strDirectory, bool updateFilterPath = true);
-    void UpdateButtons(void);
-    void ResetObservers(void);
+    virtual bool OnMessage(CGUIMessage& message) override;
```

Comment:
Isn't OnMessage already virtual, because first defined as virtual in class CGUIControl and adding "virtual" here is purely cosmetics (for better readability)? I thought once a method is virtual it stays virtual in all derived classes and omitting "virtual" in method's declaration does not make it non-virtual. ;-)